packages/core/src/render3/node_manipulation.ts

```diff
@@ -613,6 +613,8 @@ export function nativeInsertBefore(
   ngDevMode && ngDevMode.rendererInsertBefore++;
   if (isProceduralRenderer(renderer)) {
     renderer.insertBefore(parent, child, beforeNode, isMove);
+  } else if (isTemplateNode(parent)) {
+    parent.content.appendChild(child);
```

Comment:
Since `parent.content` is a document fragment, should we call `insertBefore` too?

```suggestion
    parent.content.insertBefore(child, beforeNode, isMove);
```

Or there is a risk that `beforeNode` might be outside of the `<template>` element?

It just seems a bit strange that we have different logic here for the template (always append) vs other elements (insert after).

Reply:
ops, good catch. This was supposed to be `insertBefore`, but I must have been copying things around too quickly 😅